numba_dppy/dppl_passes.py

```diff
@@ -38,6 +40,80 @@ def dpnp_available():
     except:
         return False
 
+rewrite_function_name_map = {"sum": (["np"], "sum"),
+                             "eig": (["linalg"], "eig")}
+
+class RewriteOverloadedFunctions(object):
+    def __init__(self, state, rewrite_function_name_map=rewrite_function_name_map):
+        self.state = state
+        self.function_name_map = rewrite_function_name_map
+
+    def run(self):
+        func_ir = self.state.func_ir
+        blocks = func_ir.blocks
+        topo_order = find_topo_order(blocks)
+
+        import numba_dppy.dpnp_glue.dpnpdecl
+        import numba_dppy.dpnp_glue.dpnpimpl
+        for label in topo_order:
+            block = blocks[label]
+            saved_arr_arg = {}
+            new_body = []
+            for stmt in block.body:
+                if isinstance(stmt, ir.Assign) and isinstance(stmt.value, ir.Expr):
+                    lhs = stmt.target.name
+                    rhs = stmt.value
+                    # replace np.func with name from map np.func
+                    if (rhs.op == 'getattr' and rhs.attr in rewrite_function_name_map):
+                        module_node = block.find_variable_assignment(rhs.value.name).value
+                        if ((isinstance(module_node, ir.Global) and
```

Comment:
A comment explaining the logic is required. When do we need to rewrite a `Global` node?

Reply:
Comment added.